app/src/main/java/by/paulk/mycalc/inputBuilder/InputBuilder.kt

```diff
@@ -24,10 +24,10 @@ class InputBuilder : IInputBuilder {
             }
             "C" -> result = ""
 
-            "=" -> if (input.matches(endsWithDigit)&& input.contains(containsOperator)){
-                result = input
+            "=" -> result = if (input.matches(endsWithDigit)&& input.contains(containsOperator)){
```

Comment:
I can't comment line 9-11 these calculations should be done only once, lets move it into constant in companion object